query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresBasedRequestHandler.java

```diff
@@ -392,27 +378,23 @@ public Observable<Row> handleRequest(
         throw new RuntimeException(ex);
       }
 
-      if (LOG.isDebugEnabled()) {
-        LOG.debug("Query results: [ {} ]", resultSet);
-      }
       // need to merge data especially for Postgres. That's why we need to track the map columns
-      return this.convert(resultSet)
-          .doOnComplete(
-              () -> {
-                long requestTimeMs = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
-                if (requestTimeMs > slowQueryThreshold) {
-                  try {
-                    LOG.warn(
-                        "Query Execution time: {} ms, sqlQuery: {}, queryRequest: {}, executionStats: {}",
-                        requestTimeMs,
-                        sql.getKey(),
-                        protoJsonPrinter.print(request),
-                        "Stats not available");
-                  } catch (InvalidProtocolBufferException ignore) {
-                    // ignore this exception
-                  }
-                }
-              });
+      return rowObservable.doOnComplete(
+          () -> {
+            long requestTimeMs = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+            if (requestTimeMs > slowQueryThreshold) {
+              try {
+                LOG.warn(
+                    "Query Execution time: {} ms, sqlQuery: {}, queryRequest: {}, executionStats: {}",
+                    requestTimeMs,
+                    sql.getKey(),
+                    protoJsonPrinter.print(request),
```

Comment:
So _this_ is where you want that if check. Your arguments are doing the work rather than the logger. Given that this is only triggered on threshold and log level is warn, may not be worth doing, but thats the difference from the other places.

Reply:
Got it :-)